### PR TITLE
Support AiiDA 2.0

### PR DESCRIPTION
**Pull request by ltalirz**

### CODE REVIEW

**Comment by csadorf** [comment]
@ltalirz What is the current status on this? Are there any blockers? What can we do to help move this forward?

**Comment by ltalirz** [comment]
if all the prerequisites from https://github.com/aiidalab/aiidalab-docker-stack/issues/240 are checked, then please update this PR and go ahead with merging

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM aiidateam/aiida-core:1.6.5-bionic
+# develop from feb 24th 2022
+FROM aiidateam/aiida-core@sha256:3b17bdcbb9f320aa4fe9afe8ff2248bfc07181e2ae9ac44790cf4c116a495f0b
```
Comment (csadorf) [inline Dockerfile:2]:
```suggestion
FROM aiidateam/aiida-core:2.0.1
```

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,8 @@
-aiidalab==21.11.1
-aiidalab-widgets-base==1.1.1
+#aiidalab==21.11.1
+git+https://github.com/ltalirz/aiidalab@41c362318b01aff27162fe4a9610962a01ed2a5e#egg=aiidalab
```
Comment (csadorf) [inline requirements.txt:2]:
```suggestion
aiidalab==22.6.0
```

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,8 @@
...
+#aiidalab-widgets-base==1.1.1
+git+https://github.com/aiidalab/aiidalab-widgets-base@e44c9434e97775c9d997bdd94134c6214e20a935#egg=aiidalab-widgets-base
```
Comment (csadorf) [inline requirements.txt:4]:
```suggestion
aiidalab-widgets-base==1.3.3
```

**Comment by csadorf** [comment]
The tests are failing, because the [init script tries to run `reentry scan`](https://github.com/aiidalab/aiidalab-docker-stack/runs/6967180634?check_suite_focus=true#step:8:107) which is no longer required for AiiDA 2.0. There might be other issues that need to be addressed. I can try to take care of that however it will have to wait until later this week.

@yakutovicha @ltalirz Would one of you be available to look into addressing the test failures? Some of those might need to be addressed in upstream packages [aiidateam/aiida-core](https://github.com/aiidateam/aiida-core) and/or [aiidateam/aiida-prerequisites](https://github.com/aiidateam/aiida-prerequisites).

**Comment by ltalirz** [comment]
Not sure but could this be an issue of the build cache on github actions?
https://github.com/aiidalab/aiidalab-docker-stack/blob/9639af132ee64c99ed08f7bac6dc54e3e45ad349/.github/workflows/build_and_test_on_pr.yml#L28-L34

aiida-core no longer contains "reentry scan" anywhere https://github.com/aiidateam/aiida-core/search?q=%22reentry+scan%22

**Comment by ltalirz** [comment]
No reentry here https://github.com/aiidateam/aiida-core/blob/main/.docker/opt/configure-aiida.sh

**Comment by ltalirz** [comment]
I don't get it though since we are explicitly requesting a recent aiida-core image version.
And the code was already gone in v2.0.1 https://github.com/aiidateam/aiida-core/blob/v2.0.1/.docker/opt/configure-aiida.sh

Perhaps a wrong image was pushed to aiida-core:2.0.1?

**Comment by ltalirz** [comment]
Nope
```
$ docker run -it  aiidateam/aiida-core:2.0.1 bash
root@21b8efe86b3c:/# grep reentry /opt/configure-aiida.sh
root@21b8efe86b3c:/#
```

Cache then maybe? @yakutovicha can you give a look?

**Comment by csadorf** [comment]
It's probably because the aiidalab package runs [reentry scan during the app (dependency) installation](https://github.com/aiidalab/aiidalab/blob/1200bfaec8026a500d7bfbfabea31786a33a795c/aiidalab/app.py#L274-L277). @yakutovicha Can you create an aiida-2.0 branch for the aiidalab package that requires aiida-core 2.x and where this is removed?

**Comment by yakutovicha** [comment]
I will have a look, thanks for the information.

**Comment by csadorf** [comment]
> I will have a look, thanks for the information.

Great, thank you. We can keep track of it here: https://github.com/aiidalab/aiidalab/issues/287

**Comment by ltalirz** [comment]
closing in favor of https://github.com/aiidalab/aiidalab-docker-stack/pull/254#issuecomment-1220513357

**Pull request closed without merging**